ProcessMaker/processmaker
# Resolve 'Not authorized to update user' error when creating new script

This PR resolves an issue where users with only Project Permissions were encountering a `Not authorized to update this user` error when creating a new script. The error was caused by an API request fetching the admin user and setting them as the default 'Run as User' in the Create Script Modal.

## Solution
The solution involves refactoring how the admin user is set as the default 'Run as User.' Instead of an additional API request, a query `User::where('name', 'admin')->first();` is used to retrieve the full user object. This user object is stored in a variable `runAsUserDefault` and passed into the Create Script Modal component.

## Changes
Refactored the default 'Run as User' selection to avoid additional API requests, ensuring compatibility with users having only Project Permissions.

## How to Test

**Testing Projects**

1. Ensure you have a user with only Project permissions.
2. Login with that user.
3. Navigate to Designer > Projects.
4. Click '+ Asset.'
5. Create a new script.
6. Verify no errors occur during this process.
7. Select the 'Advanced options' accordion.
8. Verify the admin user is selected by default.
9. Create the script.
10. Verify the script is created with no errors.


**Testing Scripts**

1. Login as a user with all permissions or script permissions.
2. Navigate to Designer > Script.
3. Create a new script.
4. Select the 'Advanced options' accordion.
5. Verify the admin user is selected by default.
6. Create the script.
7. Verify the script is created with no errors.

## Related Tickets & Packages
- [FOUR-12744](https://processmaker.atlassian.net/browse/FOUR-12744)
- [Projects PR](https://github.com/ProcessMaker/package-projects/pull/89)

ci:deploy
ci:next
ci:package-projects:observation/FOUR-12744

## Code Review Checklist


[FOUR-12744]: https://processmaker.atlassian.net/browse/FOUR-12744?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
